include/ttmlir/Dialect/TTIR/Transforms/Passes.td

```diff
@@ -112,4 +112,11 @@ def TTIRLoadSystemDesc: Pass<"ttir-load-system-desc", "::mlir::ModuleOp"> {
     ];
 }
 
+def TTIRBroadcastFold: Pass<"ttir-broadcast-fold", "::mlir::ModuleOp"> {
```

Comment:
We are planning to add canonicalizer soon, but I'm not sure folding framework would help here. You would have to implement fold logic for each operation which supports implicit broadcast right?

Reply:
Nevermind you can actually do this with folder. Might be better alternative then separate pass